pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -305,6 +351,20 @@ func (t *ResolvedPipelineTask) Skip(facts *PipelineRunFacts) TaskSkipStatus {
 	return facts.SkipCache[t.PipelineTask.Name]
 }
 
+// skipBecauseCELExpressionsEvaluatedToFalse confirms that the CEL when expressions have completed evaluating, and
+// it returns true if any of the CEL when expressions evaluate to false
+func (t *ResolvedPipelineTask) skipBecauseCELExpressionsEvaluatedToFalse(facts *PipelineRunFacts) bool {
+	if t.checkParentsDone(facts) {
+		for _, we := range t.PipelineTask.When {
+			if we.CEL != "" && !t.EvaluatedCEL[we.CEL] {
+				return true
+			}
+		}
+		return false
+	}
+	return false
+}
+
```

Comment:
This should be folded into `t.skipBecauseWhenExpressionsEvaluatedToFalse` because CEL expressions are part of When Expressions -- we can do this by updating [`w.isTrue`](https://github.com/Yongxuanzhang/pipeline/blob/main/pkg/apis/pipeline/v1/when_types.go#L56) then existing functionality should be reused

Reply:
Thanks! I have merged them into the isTrue. Plz check if that works.